contractcourt/chain_arbitrator.go

```diff
@@ -1287,6 +1287,10 @@ func (c *ChainArbitrator) FindOutgoingHTLCDeadline(scid lnwire.ShortChannelID,
 			continue
 		}
 
+		// Make sure the channel arbitrator has the latest view of its
+		// active HTLCs.
+		channelArb.updateActiveHTLCs()
```

Comment:
Shouldn't this always be the case as when we revoke we update the in memory HTLC set, and the latest is read in during restart?

Reply:
yeah for the channel in the process of force closing yes. But here we are reading its upstream channel, not the FCed channel, so in an edge case shown in this [test](https://github.com/lightningnetwork/lnd/pull/8751/commits/ed56a94c98127a6b705572d54fcdb8af8f2f3d1a#diff-235db90fd6236fbbc8d9ae9fd7a5d50bea961ae10d6867b90ec9a4510ba01e8eL720), the upstream channel would have no active htlcs.

I think for an active channel, we only `updateActiveHTLCs` here in `stateStep`, https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L935-L939

which is called inside `advanceState` and managed inside `channelAttendant`, which is an event loop that can be triggered via,
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L2770
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L2802
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L2816
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L2842
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L2914
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L2984
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L3045
https://github.com/lightningnetwork/lnd/blob/87d5170dec36dba1365e7b50cc1415d9c690667b/contractcourt/channel_arbitrator.go#L3068

And an active channel won't update its htlcs unless a new block comes in, which is why if we mined a block in the test, it will pass.